### PR TITLE
Added metadata to band.py for Band Plots

### PR DESCRIPTION
**Pull request by BhaswatiRoy**
ref: https://github.com/bokeh/bokeh/issues/11765

Metadata of Band Plots required some improvements for better detailing, and inclusion of API references.

### CODE REVIEW

File: examples/basic/annotations/band.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
-''' A demonstration of the Band annotation
+''' A interactive numerical band plot based on Simple Python array of data.
```
Comment (mosc9575) [inline examples/basic/annotations/band.py]:
```suggestion
''' An interactive numerical band plot based on simple Python array of data.
```

File: examples/basic/annotations/band.py [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
+    It is a combination of scatter plots and line plots added with a band of covered area.
+    The line passes through the mean of the area covered by the Band
```
Comment (mosc9575) [inline examples/basic/annotations/band.py]:
```suggestion
    The line passes through the mean of the area covered by the Band.
```

**Comment by BhaswatiRoy** [comment]
Hello @mosc9575 
Thanks for the suggestions, I have made the changes

File: examples/basic/annotations/band.py [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
+''' An interactive numerical band plot based on simple Python array of data.
+    It is a combination of scatter plots and line plots added with a band of covered area.
+    The line passes through the mean of the area covered by the Band.
 
 .. bokeh-example-metadata::
-    :apis: bokeh.models.Band, bokeh.models.ColumnDataSource
+    :apis: bokeh.models.Band, bokeh.models.ColumnDataSource, bokeh.plotting.figure, bokeh.io.show,
     :refs: :ref:`ug_basic_annotations_bands`
-    :keywords: band
+    :keywords: bandplot, figure, scatter, line, Band, add_layout, rolling
```
Comment (bryevdv) [inline examples/basic/annotations/band.py]: I'd suggest paring this down a bit:
```suggestion
    :keywords: figure, scatter, line, band, layout
```
Unfortunately we don't yet have any actual standards for what goes in "keywords" but so far they have usually been more general than specific APIs. @tcmetzger maybe we can sit down sometime soon and actually try to define style guides for keyords?

File: examples/basic/annotations/band.py [hunk 4]
```diff
@@ -1,9 +1,11 @@
-''' A demonstration of the Band annotation
+''' An interactive numerical band plot based on simple Python array of data.
+    It is a combination of scatter plots and line plots added with a band of covered area.
+    The line passes through the mean of the area covered by the Band.
```
Comment (bryevdv) [inline examples/basic/annotations/band.py]:
```suggestion
    The line passes through the mean of the area covered by the band.
```

**Review by bryevdv — CHANGES_REQUESTED** [review]
Thanks @BhaswatiRoy just a few suggestions!

**Comment by BhaswatiRoy** [comment]
Thanks @bryevdv for the suggestions, I have made the changes

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
Thanks @BhaswatiRoy !

(And thanks @mosc9575 for the review, BTW would you like to be added to @bokeh/dev?)

**Comment by mosc9575** [comment]
@bryevdv Sure. That would be an honor for me.

**Comment by bryevdv** [comment]
invite sent!

**Pull request merged**